### PR TITLE
Add C# 9 test cases for rules starting with X, V and U

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Related to #3668

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/VariableUnused.CSharp9.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Pair unusedPair = new(a, b); // Noncompliant
+
+void ParentMethod() => Invoke(static (x, y) => x++);
+
+void Invoke(Action<int, int> action) => action(1, 2);
+
+record C(string Foo, string Bar);
+
+record Pair(C A, C B);
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/VariableUnused.CSharp9.cs:16]: can you also verify inside a record method?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/XmlExternalEntityShouldNotBeParsed_XmlDocument_CSharp9.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System.Xml;
+using System.Xml.Resolvers;
+
+XmlDocument xmlDocument = new(); // Compliant - default constructor is safe in .Net 5
+
+XmlDocument doc1 = new() { XmlResolver = new XmlPreloadedResolver() }; // Compliant - FN
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/XmlExternalEntityShouldNotBeParsed_XmlDocument_CSharp9.cs:6]: to fix this, we should have a yellow _Implement_ value in the spreadsheet for XmlExternalEntityShouldNotBeParsed, right? Target typed new doesn't work in this case

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM, I've left some suggestions before merging

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/VariableUnused.CSharp9.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+
+string unusedString = string.Empty; // Noncompliant
+
+C a = new C("Foo", "1.0");
+C b = new C("Qux", "1.0");
+
+Pair unusedPair = new(a, b); // Noncompliant
+
+void ParentMethod() => Invoke(static (x, y) => x++);
```
[the remark this answers is not in the document]
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/VariableUnused.CSharp9.cs:10]: Yes. I will add a testcase.

**Pull request merged**